internal/cli/cmd/connectivity.go

```diff
@@ -76,11 +77,17 @@ func newCmdConnectivityTest(hooks Hooks) *cobra.Command {
 				return err
 			}
 
-			ctx, _ := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
+			ctx, _ := signal.NotifyContext(cmd.Context(), os.Interrupt, syscall.SIGTERM)
+
+			if params.Timeout > 0 {
+				timeoutCtx, cancelCtx := context.WithTimeoutCause(ctx, params.Timeout, fmt.Errorf("connectivity test suite timeout (%s) reached", params.Timeout))
+				defer cancelCtx()
+				ctx = timeoutCtx
+			}
 
 			go func() {
 				<-ctx.Done()
-				cc.Log("Interrupt received, cancelling tests...")
+				cc.Logf("Cancellation request (%s) received, cancelling tests...", context.Cause(ctx))
```

Comment:
In case of an interrupt, we might lose a little bit of detail in the log. This is because `signal.NotifyContext` doesn't provide the actual cause. But this should be fixed once https://github.com/golang/go/issues/60756 is fixed.

For the time being we would have to live with what `context.Cause(ctx)` provides -> `ctx.Error()`

Interrupt: `Cancellation request (context canceled) received, cancelling tests...`
Timeout: `Cancellation request (connectivity test suite timeout (5s) reached) received, cancelling tests...`